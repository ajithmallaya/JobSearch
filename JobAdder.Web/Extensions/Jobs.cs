using System.Collections.Generic;
using System.Linq;
using JobAdder.Web.Models;

namespace JobAdder.Web.Extensions
{
    public static class Jobs
    {
        public static List<Job> Filter(this IEnumerable<Job> jobs, string roles, string skills,IEnumerable<Candidate> candidates)
        {
            var filteredJobs = new List<Job>();
            foreach (var job in jobs.ToList())
            {
                var skillTags = job.Skills.Split(',');
                if (job.Name.ToLower().Contains(roles.ToLower()) || skillTags.Any( x => x.Trim().ToLower().Contains(skills.ToLower())))
                {
                    job.Candidates = candidates?.Where(x => x.Skills.Any(y => y.Trim().ToLower().Contains(skills.ToLower()))).ToList();
                    filteredJobs.Add(job);
                }
            }
            return filteredJobs;
        }
    }
}