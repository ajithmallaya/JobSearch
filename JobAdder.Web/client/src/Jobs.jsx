import React, { Component } from "react";
import "./App.css";
import axios from 'axios';
import {getAppDataById} from './helpers/appDataUtil';

class Jobs extends Component {
  constructor(props) {
    super(props);

    this.state = {
      jobs: [],
    };
  }  

  FetchJobs(roles,searchkey) {
    var that = this;
    axios.get('http://localhost:3000/Jobs/' + roles + '/' + searchkey).then(function (result) {
     console.log(result.data.result.JobWithCandidates);
      that.setState({ jobs: result.data.result.JobWithCandidates });
    });
  }   

  render() {
    return (
      <React.Fragment>
        <span className="jobs job-header"> Search Jobs </span>
        <div className="input-section">
            <div> <span>Roles: <input id="searchInputRolesText" type="text"></input> </span>
             <span>Skills: <input id="searchInputSkillsText" type="text"></input></span>
            <button onClick={ () => this.FetchJobs(getAppDataById("searchInputRolesText"),getAppDataById("searchInputSkillsText"))}> Search </button>
            </div>
            
            
        </div>
        {
		  this.state.jobs.length > 0 && 
          <div className="jobs__item">
             <div className="jobs">
			 
                 <div className="flex job-header">
                   <div className="col-3">Job</div>
                   <div className="col-3">Candidates</div>
                 </div>
                 
                {
                  this.state.jobs.map((items, index) =>
                    <div key={`${items}${index}`} className="jobs-section">
                        <div className="flex">
                          <div className="jobs col-3">
                          <span className="text">
                           <span className="label-header"> Name: </span>
                            {items.Name} </span>
                            <span className="text"> 
                            <span className="label-header">Company: </span>
                            {items.Company}  </span>
                            <span className="text">
                            <span className="label-header">Required Skills:</span>
                            {items.Skills} </span>
                          </div>
                          <div className="candidates col-3">
                            {items.Candidates.map((candidate, index) =>
                            <div key={`${candidate}${index}`} className="candidates-section">
                              <div className="flex">
                              <div className="jobs col-3">
                              <span className="text">
                              <span className="label-header">Name: </span>
                               {candidate.Name} </span>
                              <span className="text">
                              <span className="label-header"> Skills declared: </span>
                              {candidate.SkillTags} </span>
                              </div>
                              </div>
                             </div>
                          
                          )}
                        </div>
                  </div>
                  </div>
                )
              }
              </div>
            </div>
	  	}
      </React.Fragment>
    );
  }
}

export default Jobs;