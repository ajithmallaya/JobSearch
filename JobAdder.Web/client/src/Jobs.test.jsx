import { shallow } from 'enzyme';
import * as React from 'react';
import Jobs from './Jobs';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
describe('Jobs SnapShot', () => {

  it('should match snapshot', () => {
      const wrapper = shallow(<Jobs />);
      expect(wrapper.find('div')).toMatchSnapshot();
  });

});
