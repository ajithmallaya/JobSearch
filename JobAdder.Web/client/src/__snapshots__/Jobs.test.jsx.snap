// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobs SnapShot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <Jobs />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span
            className="jobs job-header"
>
             Search Jobs 
</span>,
          <div
            className="input-section"
>
            <div>
                         
                        <span>
                                    Roles: 
                                    <input
                                                id="searchInputRolesText"
                                                type="text"
                                    />
                                     
                        </span>
                        <span>
                                    Skills: 
                                    <input
                                                id="searchInputSkillsText"
                                                type="text"
                                    />
                        </span>
                        <button
                                    onClick={[Function]}
                        >
                                     Search 
                        </button>
            </div>
</div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Search Jobs ",
            "className": "jobs job-header",
          },
          "ref": null,
          "rendered": " Search Jobs ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
               
              <span>
                            Roles: 
                            <input
                                          id="searchInputRolesText"
                                          type="text"
                            />
                             
              </span>
              <span>
                            Skills: 
                            <input
                                          id="searchInputSkillsText"
                                          type="text"
                            />
              </span>
              <button
                            onClick={[Function]}
              >
                             Search 
              </button>
</div>,
            "className": "input-section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                <span>
                  Roles: 
                  <input
                                    id="searchInputRolesText"
                                    type="text"
                  />
                   
</span>,
                <span>
                  Skills: 
                  <input
                                    id="searchInputSkillsText"
                                    type="text"
                  />
</span>,
                <button
                  onClick={[Function]}
>
                   Search 
</button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Roles: ",
                    <input
                      id="searchInputRolesText"
                      type="text"
/>,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Roles: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "id": "searchInputRolesText",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  " ",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Skills: ",
                    <input
                      id="searchInputSkillsText"
                      type="text"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Skills: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "id": "searchInputSkillsText",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Search ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": " Search ",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <span
              className="jobs job-header"
>
               Search Jobs 
</span>,
            <div
              className="input-section"
>
              <div>
                             
                            <span>
                                          Roles: 
                                          <input
                                                        id="searchInputRolesText"
                                                        type="text"
                                          />
                                           
                            </span>
                            <span>
                                          Skills: 
                                          <input
                                                        id="searchInputSkillsText"
                                                        type="text"
                                          />
                            </span>
                            <button
                                          onClick={[Function]}
                            >
                                           Search 
                            </button>
              </div>
</div>,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Search Jobs ",
              "className": "jobs job-header",
            },
            "ref": null,
            "rendered": " Search Jobs ",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                 
                <span>
                                Roles: 
                                <input
                                                id="searchInputRolesText"
                                                type="text"
                                />
                                 
                </span>
                <span>
                                Skills: 
                                <input
                                                id="searchInputSkillsText"
                                                type="text"
                                />
                </span>
                <button
                                onClick={[Function]}
                >
                                 Search 
                </button>
</div>,
              "className": "input-section",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span>
                    Roles: 
                    <input
                                        id="searchInputRolesText"
                                        type="text"
                    />
                     
</span>,
                  <span>
                    Skills: 
                    <input
                                        id="searchInputSkillsText"
                                        type="text"
                    />
</span>,
                  <button
                    onClick={[Function]}
>
                     Search 
</button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Roles: ",
                      <input
                        id="searchInputRolesText"
                        type="text"
/>,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Roles: ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "id": "searchInputRolesText",
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    " ",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Skills: ",
                      <input
                        id="searchInputSkillsText"
                        type="text"
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Skills: ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "id": "searchInputSkillsText",
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Search ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": " Search ",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          false,
        ],
        "type": Symbol(react.fragment),
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
      "attachTo": undefined,
      "hydrateIn": undefined,
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
         
        <span>
                Roles: 
                <input
                        id="searchInputRolesText"
                        type="text"
                />
                 
        </span>
        <span>
                Skills: 
                <input
                        id="searchInputSkillsText"
                        type="text"
                />
        </span>
        <button
                onClick={[Function]}
        >
                 Search 
        </button>
</div>,
      "className": "input-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          <span>
            Roles: 
            <input
                        id="searchInputRolesText"
                        type="text"
            />
             
</span>,
          <span>
            Skills: 
            <input
                        id="searchInputSkillsText"
                        type="text"
            />
</span>,
          <button
            onClick={[Function]}
>
             Search 
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Roles: ",
              <input
                id="searchInputRolesText"
                type="text"
/>,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Roles: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "searchInputRolesText",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Skills: ",
              <input
                id="searchInputSkillsText"
                type="text"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Skills: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "searchInputSkillsText",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Search ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Search ",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
           
          <span>
                    Roles: 
                    <input
                              id="searchInputRolesText"
                              type="text"
                    />
                     
          </span>
          <span>
                    Skills: 
                    <input
                              id="searchInputSkillsText"
                              type="text"
                    />
          </span>
          <button
                    onClick={[Function]}
          >
                     Search 
          </button>
</div>,
        "className": "input-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            <span>
              Roles: 
              <input
                            id="searchInputRolesText"
                            type="text"
              />
               
</span>,
            <span>
              Skills: 
              <input
                            id="searchInputSkillsText"
                            type="text"
              />
</span>,
            <button
              onClick={[Function]}
>
               Search 
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Roles: ",
                <input
                  id="searchInputRolesText"
                  type="text"
/>,
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Roles: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "searchInputRolesText",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              " ",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Skills: ",
                <input
                  id="searchInputSkillsText"
                  type="text"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Skills: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "searchInputSkillsText",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Search ",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": " Search ",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          <span>
            Roles: 
            <input
                        id="searchInputRolesText"
                        type="text"
            />
             
</span>,
          <span>
            Skills: 
            <input
                        id="searchInputSkillsText"
                        type="text"
            />
</span>,
          <button
            onClick={[Function]}
>
             Search 
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Roles: ",
              <input
                id="searchInputRolesText"
                type="text"
/>,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Roles: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "searchInputRolesText",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Skills: ",
              <input
                id="searchInputSkillsText"
                type="text"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Skills: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "searchInputSkillsText",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Search ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Search ",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
